stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1593,5 +1637,46 @@ public class ChatClient internal constructor(
                 defaultNotificationConfig
             )
         }
+
+        /**
+         * Handles remote message.
+         * If user is not connected - automatically restores last user credentials and sets user without connecting to the socket.
+         * Remote message will be handled internally unless user overrides [ChatNotificationHandler.onFirebaseMessage]
+         * Be sure to initialize ChatClient before calling this method!
+         *
+         * @see [ChatNotificationHandler.onFirebaseMessage]
+         * @throws IllegalStateException if called before initializing ChatClient
+         */
+        @Throws(IllegalStateException::class)
+        public fun handleRemoteMessage(remoteMessage: RemoteMessage) {
+            ensureClientInitialized()
+            instance!!.run {
+                setUserWithoutConnectingIfNeeded()
+                notifications.onFirebaseMessage(remoteMessage, pushNotificationReceivedListener)
+            }
+        }
+
+        @Throws(IllegalStateException::class)
+        internal suspend fun displayNotificationWithData(channelType: String, channelId: String, messageId: String) {
+            ensureClientInitialized()
+            instance!!.notifications.displayNotificationWithData(channelId, channelType, messageId)
+        }
+
+        /**
+         * Sets Firebase token.
+         * Be sure to initialize ChatClient before calling this method!
+         *
+         * @throws IllegalStateException if called before initializing ChatClient
+         */
+        @Throws(IllegalStateException::class)
+        public fun setFirebaseToken(token: String) {
+            ensureClientInitialized()
+            instance!!.notifications.setFirebaseToken(token)
+        }
+
+        @Throws(IllegalStateException::class)
+        private fun ensureClientInitialized() {
```

Comment:
What about to return the instance if it is initialized? :thinking: 
Then, we don't need to use `instance!!` on the rest of methods

Reply:
nice one. Thanks!